t/proxyunits.t

```diff
@@ -69,9 +69,44 @@ for my $msrv (@mocksrvs) {
     push(@mbe, $be);
 }
 
-# Basic test with no backends. Write a command to the client, and check the
-# response.
+# $ps_send : request to proxy
+# $be_recv_ref : ref to a hashmap from be index to an array of received requests for validation.
+# $be_send_ref : ref to a hashmap from be index to an array of responses to proxy.
+# $ps_recv_ref : ref to response returned by proxy
+sub proxy_test {
+    my ($ps_send, $be_recv_ref, $be_send_ref, $ps_recv_ref) = @_;
+
+    my %be_recv = %{$be_recv_ref};
```

Comment:
skip these. just keep the reference and deref it properly below.

ie `keys %{$be_recv}`
`for my $recv (@{$be_recv->{$idx}})` (arrow deref)
etc.

Reply:
Great, still learning perl.

Reply:
Was attempting to structure things so you wouldn't need to learn much perl to write a bunch of tests :) Oh well. Sorry.